src/components/calculator.js

```diff
@@ -1,33 +1,55 @@
 import './Calculator.css';
+import { useState } from 'react';
 import Text from './calculatorText';
+import calculate from '../Logic/calculate';
 
-const CalculatorElement = () => (
-  <>
-    <Text />
-    <div className="grid-container">
-      <input className="input-number" value={0} />
-      <button type="button">AC</button>
-      <button type="button">+/-</button>
-      <button type="button">%</button>
-      <button type="button">/</button>
-      <button type="button">7</button>
-      <button type="button">8</button>
-      <button type="button">9</button>
-      <button type="button">X</button>
-      <button type="button">4</button>
-      <button type="button">5</button>
-      <button type="button">6</button>
-      <button type="button">-</button>
-      <button type="button">1</button>
-      <button type="button">2</button>
-      <button type="button">3</button>
-      <button type="button">+</button>
-      <button type="button" className="item-seventeen">0</button>
-      <button type="button" className="item-eighteen">.</button>
-      <button type="button">=</button>
-    </div>
+const CalculatorElement = () => {
+  const [state, setState] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
 
-  </>
-);
+  const handleButtonClicks = (buttonValue) => {
+    const outPut = calculate(state, buttonValue);
+    setState(outPut);
+    // console.log(outPut);
+  };
+
+  return (
+    <>
+      <Text />
+      <div className="grid-container">
+        <div className="input-number">
+          {state.total}
+          {' '}
+          {state.operation}
+          {' '}
+          {state.next}
+        </div>
+        <button type="button" value="AC" onClick={(e) => handleButtonClicks(e.target.value)}>AC</button>
+        <button type="button" value="+/-" onClick={(e) => handleButtonClicks(e.target.value)}>+/-</button>
+        <button type="button" value="%" onClick={(e) => handleButtonClicks(e.target.value)}>%</button>
+        <button type="button" value="/" onClick={(e) => handleButtonClicks(e.target.value)}>/</button>
```

Comment:
- [ Required ] I think it would be better if you ensure that the division operation works as expected. Currently, when I try to perform a division operation I get the error below. I think this is because the value on the division button should be `÷` while the value on your button is `/`. Please see the screenshot below and rectify this.

## Project requirements
![Screenshot from 2023-09-14 23-51-59](https://github.com/Ayokunnumi1/math-magician/assets/16392046/f064d736-eaac-4361-96c0-0ab6d0fb5ebd)